### ISSUE TITLE
modularize spamfilter

### ISSUE THREAD

**Issue by kaniini** [rabbitircd/rabbitircd#2]
topic pretty much covers it.  you'll probably just want to rewrite the whole thing when you refactor this.

**Comment by jdhore** [comment]
For the record, for whoever does this (if it's not @kaniini ), @kaniini and others did some research a few years ago and Oniguruma was the fastest regex engine out there, though you'd probably want to make spamfilter use PCRE as it's more common and may be faster now that PCRE has JIT.

**Comment by kaniini** [comment]
yeah, TRE is awful and should go away.  but we're stuck with TRE for now for backwards compat with unreal 3.2.

**Comment by kaniini** [comment]
moving this to 0.2
